crates/uv-client/src/html.rs

```diff
@@ -225,6 +226,57 @@ impl SimpleHtml {
     }
 }
 
+/// A parsed structure from PyPI "HTML" index format listing all available packages.
+#[derive(Debug, Clone)]
+pub(crate) struct SimpleIndexHtml {
+    /// The list of project names available in the index.
+    pub(crate) projects: Vec<PackageName>,
+}
+
+impl SimpleIndexHtml {
+    /// Parse the list of project names from the Simple API index HTML page.
+    pub(crate) fn parse(text: &str) -> Result<Self, Error> {
+        let dom = tl::parse(text, tl::ParserOptions::default())?;
+
+        // Parse each `<a>` tag to extract the project name.
+        let parser = dom.parser();
+        let mut projects = dom
+            .nodes()
+            .iter()
+            .filter_map(|node| node.as_tag())
+            .filter(|link| link.name().as_bytes() == b"a")
+            .filter_map(|link| Self::parse_anchor_project_name(link, parser))
+            .filter_map(|name| PackageName::from_str(&name).ok())
+            .collect::<Vec<_>>();
+
+        // Sort for deterministic ordering.
+        projects.sort_unstable();
+
+        Ok(Self { projects })
+    }
+
+    /// Parse a project name from an `<a>` tag.
+    ///
+    /// Returns `None` if the `<a>` doesn't have an `href` attribute or text content.
+    fn parse_anchor_project_name(link: &HTMLTag, parser: &tl::Parser) -> Option<SmallString> {
+        // Extract the href.
+        link.attributes()
+            .get("href")
+            .flatten()
+            .filter(|bytes| !bytes.as_bytes().is_empty())?;
```

Comment:
What does pip do here, should we check that the href is some form of `/<project>/`, to remove non-project links?